src/nvim/highlight.c

```diff
@@ -753,11 +753,19 @@ int hl_blend_attrs(int back_attr, int front_attr, bool *through)
 
     cattrs.rgb_ae_attr &= ~HL_BG_INDEXED;
   }
-  cattrs.rgb_bg_color = rgb_blend(ratio, battrs.rgb_bg_color,
-                                  fattrs.rgb_bg_color);
 
-  cattrs.hl_blend = -1;  // blend property was consumed
+  HlAttrs original_back = syn_attr2entry(back_attr);
```

Comment:
We've already looked up these once so it's a bit weird to need to look up them again. perhaps rewrite the original lookup to 
```
HlAttrs battrs_raw = syn_attr2entry(back_attr);
HlAttrs battrs = get_colors_force(battrs_raw);
```
`get_colors_force` in only used for blending so it can be changed.